app/Commands/Compile.hs

```diff
@@ -0,0 +1,178 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Commands.Compile where
+
+import Commands.Extra
+import Data.ByteString qualified as BS
+import Data.FileEmbed qualified as FE
+import Data.Text.IO qualified as TIO
+import MiniJuvix.Prelude hiding (Doc)
+import Options.Applicative
+import System.Environment
+import System.Process qualified as P
+
+minijuvixBuildDir :: FilePath
+minijuvixBuildDir = ".minijuvix-build"
```

Comment:
Intuitively I would say that this directory should be `<proj-root>/.minijuvix-build`. Do we have a reason not to do that?